### ISSUE TITLE
Cannot click on organization switcher when shadcn Sheet is open

### ISSUE THREAD

**Issue by gunwooterry** [clerk/javascript#3739]
### Preliminary Checks

- [X] I have reviewed the documentation: https://clerk.com/docs

- [X] I have searched for existing issues: https://github.com/clerk/javascript/issues

- [X] I have not already reached out to Clerk support via email or Discord (if you have, no need to open an issue here)

- [X] This issue is not a question, general help request, or anything other than a bug report directly related to Clerk. Please ask questions in our Discord community: https://clerk.com/discord.

### Reproduction

https://github.com/gunwooterry/clerk-issue-reproduction

### Publishable key

pk_test_cmVndWxhci1raXR0ZW4tNjAuY2xlcmsuYWNjb3VudHMuZGV2JA

### Description

Clerk's premade component, `OrganizationSwitcher`, should be placed on top of anything. Its z-index is set to 10000.
However, when used with a `Sheet` from [shadcn](https://ui.shadcn.com/docs/components/sheet), one of the widely used UI component library, the organization switcher popover cannot be clicked. It's not a simple z-index problem, as it is not blocked by other components visually. There are no other components with a higher z-index than the popover.
It also cannot be selected using the Chrome inspector, as shown in the screenshots:
There is a deployed example of reproduction -- https://clerk-issue-reproduction.vercel.app/ -- so please refer to it.

<img width="389" alt="Screenshot 2024-07-17 at 6 10 31 PM" src="https://github.com/user-attachments/assets/48c078aa-ae63-41b2-90aa-24f0c4920e18">
<img width="388" alt="Screenshot 2024-07-17 at 6 10 51 PM" src="https://github.com/user-attachments/assets/dab9eca0-3ac5-4914-b649-65f6446b9f7e">

**Steps to reproduce:**
1. Place `OrganizationSwitcher` on a shadcn Sheet (https://ui.shadcn.com/docs/components/sheet)
2. Click on the organization switcher to open the popover
3. No components on the switcher popover can be clicked

**Expected behavior:**
Components in the popover (e.g. switching organizations, organization settings) should be clicked.

**Actual behavior:**
No click events are delivered to the popover.

### Environment

```Shell
System:
    OS: macOS 14.5
    CPU: (10) arm64 Apple M1 Pro
    Memory: 39.41 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 22.2.0 - /opt/homebrew/bin/node
    Yarn: 1.22.18 - /opt/homebrew/bin/yarn
    npm: 10.7.0 - /opt/homebrew/bin/npm
    pnpm: 8.15.1 - /opt/homebrew/bin/pnpm
    Watchman: 2024.06.17.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 126.0.6478.127
    Safari: 17.5
  npmPackages:
    @clerk/nextjs: ^5.2.4 => 5.2.4
    @radix-ui/react-dialog: ^1.1.1 => 1.1.1
    @radix-ui/react-slot: ^1.1.0 => 1.1.0
    @types/node: ^20 => 20.14.11
    @types/react: ^18 => 18.3.3
    @types/react-dom: ^18 => 18.3.0
    class-variance-authority: ^0.7.0 => 0.7.0
    clsx: ^2.1.1 => 2.1.1
    eslint: ^8 => 8.57.0
    eslint-config-next: 14.2.5 => 14.2.5
    lucide-react: ^0.408.0 => 0.408.0
    next: 14.2.5 => 14.2.5
    postcss: ^8 => 8.4.39
    react: ^18 => 18.3.1
    react-dom: ^18 => 18.3.1
    tailwind-merge: ^2.4.0 => 2.4.0
    tailwindcss: ^3.4.1 => 3.4.6
    tailwindcss-animate: ^1.0.7 => 1.0.7
    typescript: ^5 => 5.5.3
```

**Comment by alexcarpenter** [comment]
@gunwooterry we're looking to make improvements here with regards to where these elements get portaled into the dom which is part of the issue your experiencing when using some of the clerk components within dialog components.

A temporary solution would be to add pointer events auto to the `organizationSwitcherPopoverRootBox` element.

```tsx
<OrganizationSwitcher
  appearance={{
    elements: {
      organizationSwitcherPopoverRootBox: {
        pointerEvents: "auto",
      },
    },
  }}
/>
```

**Comment by clerk-cookie** [comment]
Hello 👋

We currently close issues after 40 days of inactivity. It's been 30 days since the last update here. If we missed this issue, please reply here. Otherwise, we'll close this issue in 10 days.

As a friendly reminder: The best way to see an issue fixed is to open a pull request. If you're not sure how to do that, please check out our [contributing guide](https://github.com/clerk/javascript/blob/main/docs/CONTRIBUTING.md).

Thanks for being a part of the Clerk community! 🙏

**Comment by clerk-cookie** [comment]
Hello again 👋

After 40 days of no activity, we'll close this issue. Keep in mind, I'm just a robot, so if I've closed this issue in error, please reply here and my human colleagues will reopen it.

As a friendly reminder: The best way to see an issue fixed is to open a pull request. If you're not sure how to do that, please check out our [contributing guide](https://github.com/clerk/javascript/blob/main/docs/CONTRIBUTING.md).

Thanks for being a part of the Clerk community! 🙏

**Comment by clerk-cookie** [comment]
This issue has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
